util/mobile_base.hpp

```diff
@@ -0,0 +1,325 @@
+    auto count = waypoints.size();
+
+    VectorXd times(count);
+    MatrixXd positions(3, count);
+    MatrixXd velocities(3, count);
+    MatrixXd accelerations(3, count);
+
+    for (auto idx = 0; idx < count; ++idx) {
+      times(idx) = waypoints.at(idx).t;
+      positions(0, idx) = waypoints.at(idx).pos.x;
+      positions(1, idx) = waypoints.at(idx).pos.y;
+      positions(2, idx) = waypoints.at(idx).pos.theta;
+
+      velocities(0, idx) = waypoints.at(idx).vel.x;
+      velocities(1, idx) = waypoints.at(idx).vel.y;
+      velocities(2, idx) = waypoints.at(idx).vel.theta;
+
+      accelerations(0, idx) = waypoints.at(idx).acc.x;
+      accelerations(1, idx) = waypoints.at(idx).acc.y;
+      accelerations(2, idx) = waypoints.at(idx).acc.theta;
+    }
+
+    return CartesianGoal(times, positions, velocities, accelerations);
+  }
+
+  const Eigen::VectorXd& times() const { return times_; }
+  const Eigen::Matrix<double, 3, Eigen::Dynamic>& positions() const { return positions_; }
+  const Eigen::Matrix<double, 3, Eigen::Dynamic>& velocities() const { return velocities_; }
+  const Eigen::Matrix<double, 3, Eigen::Dynamic>& accelerations() const { return accelerations_; }
+
+private:
+  CartesianGoal(const Eigen::VectorXd& times,
+       const Eigen::Matrix<double, 3, Eigen::Dynamic>& positions,
+       const Eigen::Matrix<double, 3, Eigen::Dynamic>& velocities,
+       const Eigen::Matrix<double, 3, Eigen::Dynamic>& accelerations)
+    : times_(times),
+      positions_(positions),
+      velocities_(velocities),
+      accelerations_(accelerations) {}
+
+  // TODO: some info here capturing above states/options
+  const Eigen::VectorXd times_{0};
+  const Eigen::Matrix<double, 3, Eigen::Dynamic> positions_{0, 0};
+  const Eigen::Matrix<double, 3, Eigen::Dynamic> velocities_{0, 0};
+  const Eigen::Matrix<double, 3, Eigen::Dynamic> accelerations_{0, 0};
+};
+
+//////////////////////////////
+
+class MobileBase {
+
+public:
+
+  // Parameters for creating a base
+  struct Params: public GroupManager::Params {
+  };
+
+  // This is a generic implementation that assumes the use of a wheel-space trajectory
+  bool update() {
+    // updates dt and last feedback message
+    auto ret = group_manager_->update();
+    // now update odometry
+    auto vel = group_manager_->lastFeedback().getVelocity();
+    updateOdometry(vel, group_manager_->dT());
+
+    auto& cmd = group_manager_->pendingCommand();
+
+    while (base_trajectories_.front() && base_trajectories_.front()->getEndTime() < group_manager_->lastTime()) {
+      base_trajectories_.pop();
+    }
+
+    // go into compliance mode if no trajectory
+    if (base_trajectories_.size() == 0) {
+      auto size = group_manager_->size();
+
+      VectorXd compliantState;
+      compliantState.resize(size);
+      double nan = std::numeric_limits<double>::quiet_NaN();
+      for (auto i=0; i<size; ++i) {
+        compliantState(i) = nan;
+      }
```

Comment:
I think this should do the trick:
`VectorXd compliantState = VectorXd::Constant(1, size, std::numeric_limits<double>::quiet_NaN());`